emscripten.py

```diff
@@ -2107,17 +2107,34 @@ def create_sending_wasm(invoke_funcs, jscall_sigs, forwarded_json, metadata):
     global_vars = [] # linkable code accesses globals through function calls
 
   implemented_functions = set(metadata['implementedFunctions'])
-  global_funcs = list(set([key for key, value in forwarded_json['Functions']['libraryFunctions'].items() if value != 2]).difference(set(global_vars)).difference(implemented_functions))
+  library_funcs = set(k for k, v in forwarded_json['Functions']['libraryFunctions'].items() if v != 2)
+  global_funcs = list(library_funcs.difference(set(global_vars)).difference(implemented_functions))
 
   jscall_funcs = ['jsCall_' + sig for sig in jscall_sigs]
 
   send_items = (basic_funcs + invoke_funcs + jscall_funcs + global_funcs +
                 basic_vars + global_vars)
 
-  def math_fix(g):
-    return g if not g.startswith('Math_') else g.split('_')[1]
+  def fixImportName(g):
+    if g.startswith('Math_'):
+      return g.split('_')[1]
+    # Unlike fastcomp the wasm backend doesn't use the '_' prefix for native
+    # symbols.  Emscripten currently expects symbols to start with '_' so we
+    # artificially add them to the output of emscripten-wasm-finalize and them
+    # strip them again here.
+    if g.startswith('_'):
+      return g[1:]
+    return g
+
+  send_items_map = OrderedDict()
+  seen = set()
+  for name in send_items:
+    internal_name = fixImportName(name)
+    if internal_name in seen:
```

Comment:
1) This never happens because we don't update `seen`?
2) Instead of having a separate set where there can be mismatches, how about checking `if internal_name in send_items_map`

Reply:
Sorry, I copy and pasted this .. that is what I meant to do originally.